Sources/PackageLoading/PackageExtensions.swift

```diff
@@ -154,19 +154,26 @@ extension Package {
         return modules
     }
     
-    private func modulify(_ path: String, name: String) throws -> Module {
+    private func modulify(_ path: String, name: String, isTest: Bool) throws -> Module {
         let walked = walk(path, recursing: shouldConsiderDirectory).map{ $0 }
         
         let cSources = walked.filter{ isValidSource($0, validExtensions: Sources.validCFamilyExtensions) }
         let swiftSources = walked.filter{ isValidSource($0, validExtensions: Sources.validSwiftExtensions) }
         
         if !cSources.isEmpty {
             guard swiftSources.isEmpty else { throw Module.Error.mixedSources(path) }
-            return try ClangModule(name: name, sources: Sources(paths: cSources, root: path))
+
+          #if os(Linux)
+            guard !isTest else {
+                fatalError("\(name): C family test modules are not supported on linux.")
+            }
+          #endif
```

Comment:
This is unfortunate, because otherwise cross-platform packages will just fail to build. I would prefer that -- for now -- we simply automatically ignore these targets on Linux, and work on a real platform compatibility specification API (in PackageDescription).